src/coreclr/gc/gc.cpp

```diff
@@ -17796,6 +17799,9 @@ void gc_heap::balance_heaps (alloc_context* acontext)
 
 ptrdiff_t gc_heap::get_balance_heaps_uoh_effective_budget (int generation_num)
 {
+#ifdef USE_REGIONS
+    return dd_new_allocation (dynamic_data_of (generation_num));
+#else
```

Comment:
I would write it this way to avoid duplicated code -

```cplusplus
ptrdiff_t gc_heap::get_balance_heaps_uoh_effective_budget (int generation_num)
{
#ifndef USE_REGIONS
    if (heap_hard_limit)
    {
        const ptrdiff_t free_list_space = generation_free_list_space (generation_of (generation_num));
        heap_segment* seg = generation_start_segment (generation_of (generation_num));
        assert (heap_segment_next (seg) == nullptr);
        const ptrdiff_t allocated = heap_segment_allocated (seg) - seg->mem;
        // We could calculate the actual end_of_seg_space by taking reserved - allocated,
        // but all heaps have the same reserved memory and this value is only used for comparison.
        return free_list_space - allocated;
    }
    else
#endif //!USE_REGIONS
    {
        return dd_new_allocation (dynamic_data_of (generation_num));
    }
}
```